indico/modules/events/contributions/client/js/PublicationButton.jsx

```diff
@@ -0,0 +1,62 @@
+// This file is part of Indico.
+// Copyright (C) 2002 - 2021 CERN
+//
+// Indico is free software; you can redistribute it and/or
+// modify it under the terms of the MIT License; see the
+// LICENSE file for more details.
+
+import publicationURL from 'indico-url:contributions.manage_publication';
+
+import PropTypes from 'prop-types';
+import React, {useState} from 'react';
+
+import {Translate} from 'indico/react/i18n';
+import {handleAxiosError, indicoAxios} from 'indico/utils/axios';
+
+import PublicationModal from './PublicationModal';
+
+export default function PublicationButton({eventId, onSuccess}) {
+  const url = publicationURL({event_id: eventId});
+  const [modalOpen, setModalOpen] = useState(false);
+
+  const onPublish = async () => {
+    try {
+      await indicoAxios.put(url);
+      onSuccess();
+    } catch (error) {
+      handleAxiosError(error);
+    }
+  };
+
+  const trigger = <span onClick={() => setModalOpen(true)}>Publish now</span>;
+
+  const body = (
+    <>
+      <p>
+        <Translate>Are you sure you want to publish the contribution list?</Translate>
+      </p>
+      <p>
+        <Translate>By doing so the following menu items will be accessible:</Translate>
+      </p>
+    </>
+  );
+
+  return (
+    <PublicationModal
+      open={modalOpen}
+      setModalOpen={setModalOpen}
+      trigger={trigger}
+      header={Translate.string('Publish now')}
+      body={body}
+      onClick={() => {
+        onPublish();
+        setModalOpen(false);
```

Comment:
Maybe this should go into onPublish (and only run in case of success)?

Reply:
I was thinking in case of error we may want to hide the `PublicationModal` anyway?

Reply:
since we don't expect errors there and the user can just close it anyway, i'd rather not do that automatically

Reply:
ok changed :+1: